src/components/vr_cooperation/include/vr_cooperation/commands/on_service_deactivated_notification.h

```diff
@@ -35,27 +35,35 @@
 
 #include "utils/macro.h"
 #include "vr_cooperation/vr_module.h"
+#include "vr_cooperation/interface/hmi.pb.h"
 
 namespace vr_cooperation {
 
 namespace commands {
 
-class OnServiceDeactivatedNotification {
+class OnServiceDeactivatedNotification: public Command {
  public:
   /**
    * @brief OnServiceDeactivatedNotification class constructor
-   * @param parent VRModule parent
+   * @param parent VRModule parent pointer
+   * @param message GPB message
    */
-  explicit OnServiceDeactivatedNotification(VRModule* parent);
+  OnServiceDeactivatedNotification(VRModule* parent,
+                                   const vr_hmi_api::ServiceMessage& message);
 
   /**
-   * @brief Execute command
-   * @param message message to be sent to mobile
+   * @brief run command
    */
-  void Execute(const application_manager::MessagePtr& message);
+  virtual void Run();
+
+  /**
+   * @brief on timeout reaction
+   */
+  virtual void OnTimeout();
```

Comment:
OnTimeout is not necessary for notification, that's why please make another base class special for notifications